org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/managers/StandardProjectsManager.java

```diff
@@ -307,14 +317,16 @@ public static void configureSettings(Preferences preferences) {
 			});
 		}
 		JavaCore.setOptions(javaOptions);
-		new WorkspaceJob("Clean workspace...") {
+		if (cleanWorkspace && preferences.isAutobuildEnabled()) {
+			new WorkspaceJob("Clean workspace...") {
 
-			@Override
-			public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
-				ResourcesPlugin.getWorkspace().build(IncrementalProjectBuilder.CLEAN_BUILD, monitor);
-				return Status.OK_STATUS;
-			}
-		}.schedule();
+				@Override
+				public IStatus runInWorkspace(IProgressMonitor monitor) throws CoreException {
```

Comment:
Just curious, why do we need a clean build here?

Reply:
Because the JDT settings has been changed.